src/reports/library/ascii_writer.cpp

```diff
@@ -0,0 +1,49 @@
+#include <iostream>
+
+#include "ascii_writer.hpp"
+
+AsciiWriter::AsciiWriter(const std::string& report_name)
+: IoWriter(report_name), m_file(report_name) {}
+
+AsciiWriter::~AsciiWriter() {}
+
+void AsciiWriter::write(const std::string& name, float* buffer, int steps_to_write, int total_steps, int total_compartments) {
+    m_file << name << std::endl;
+    size_t buffer_size = total_steps * total_compartments;
+    for(int i=0; i<buffer_size; i++) {
+        m_file << buffer[i] << ", ";
+    }
+    m_file << std::endl << std::endl;
+}
+
+void AsciiWriter::write(const std::string& name, const std::vector<int>& buffer) {
+
+}
+
+void AsciiWriter::write(const std::string& name, const std::vector<uint32_t>& buffer) {
```

Comment:
these can probably be templetized

Reply:
I thought about that, but found it hard when you need to pass some MPI types to the functions (i.e. H5T_STD_I32LE)
I would need to check the template type to pass one or another variable...

Reply:
using templates for the standard types, and having some specializations for exotic types seems like a good way around this.  However, if the `ascii_writer` isn't needed, it's moot.